esp-hal/src/spi/slave.rs

```diff
@@ -656,71 +633,106 @@ pub trait Instance: private::Sealed {
         reg_block.user().write(|w| unsafe { w.bits(0) });
         reg_block.ctrl().write(|w| unsafe { w.bits(0) });
 
-        reg_block.slave().write(|w| w.mode().set_bit());
+        reg_block.slave().write(|w| {
+            #[cfg(esp32)]
+            w.slv_wr_rd_buf_en().set_bit();
+
+            w.mode().set_bit()
+        });
+        reset_spi(reg_block);
 
         reg_block.user().modify(|_, w| {
-            w.usr_miso_highpart().clear_bit();
             w.doutdin().set_bit();
-            w.usr_miso().clear_bit();
-            w.usr_mosi().clear_bit();
-            w.usr_dummy_idle().clear_bit();
-            w.usr_addr().clear_bit();
-            w.usr_command().clear_bit();
             w.sio().clear_bit()
         });
 
-        #[cfg(not(esp32s2))]
-        reg_block.clk_gate().modify(|_, w| {
-            w.clk_en().clear_bit();
-            w.mst_clk_active().clear_bit();
-            w.mst_clk_sel().clear_bit()
-        });
-
-        reg_block.ctrl().modify(|_, w| {
-            w.q_pol().clear_bit();
-            w.d_pol().clear_bit();
-            #[cfg(not(esp32s2))]
-            w.hold_pol().clear_bit();
-            #[cfg(esp32s2)]
-            w.wp().clear_bit();
-            w
-        });
-
+        #[cfg(not(esp32))]
         reg_block.misc().write(|w| unsafe { w.bits(0) });
     }
 
-    fn set_data_mode(&mut self, data_mode: SpiMode) -> &mut Self {
+    fn set_data_mode(&mut self, data_mode: SpiMode, dma: bool) -> &mut Self {
         let reg_block = self.register_block();
+        #[cfg(esp32)]
+        {
+            reg_block.pin().modify(|_, w| {
+                w.ck_idle_edge()
+                    .bit(matches!(data_mode, SpiMode::Mode0 | SpiMode::Mode1))
+            });
+            reg_block.user().modify(|_, w| {
+                w.ck_i_edge()
+                    .bit(matches!(data_mode, SpiMode::Mode1 | SpiMode::Mode2))
+            });
+            reg_block.ctrl2().modify(|_, w| unsafe {
+                match data_mode {
+                    SpiMode::Mode0 => {
```

Comment:
Does it make sense to "configure" Mode0 and 2 here?

Reply:
Yes, in case we add CPU-controlled mode we won't have to debug why ESP32 panics :)